src/Stripe.net/Services/Subscriptions/StripeSubscriptionService.cs

```diff
@@ -61,13 +48,12 @@ public virtual StripeSubscription Update(string subscriptionId, StripeSubscripti
                     this.SetupRequestOptions(requestOptions)));
         }
 
-        public virtual StripeSubscription Cancel(string subscriptionId, bool cancelAtPeriodEnd = false, StripeRequestOptions requestOptions = null)
+        public virtual StripeSubscription Cancel(string subscriptionId, StripeSubscriptionCancelOptions cancelOptions, StripeRequestOptions requestOptions = null)
         {
-            var url = string.Format(Urls.Subscriptions + "/{0}", subscriptionId);
-            url = ParameterBuilder.ApplyParameterToUrl(url, "at_period_end", cancelAtPeriodEnd.ToString());
-
             return Mapper<StripeSubscription>.MapFromJson(
-                Requestor.Delete(url, this.SetupRequestOptions(requestOptions)));
+                Requestor.Delete(
```

Comment:
I am a _little_ bit worried about this line. I tried testing it fully and I think it works but I could not force stripe-mock to reject my options (When sending fake parameters). It either means it does not validate GET params properly (my guess) or that something truncates the GET parameters when sent with Delete. I have spent a while on this and need a second pair of eyes to make sure.

Reply:
Flagging that with the latest stripe-mock I have been able to confirm the params would be sent properly so it seems to work fine!